File: backend/src/domain/entities/Job.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
} from 'typeorm';

@Entity('jobs')
export class Job {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'organization_id' })
  organizationId: number;

  @Column({ name: 'poster_member_id' })
  posterMemberId: number;

  @Column({ type: 'boolean', name: 'is_referral', default: false })
  isReferral: boolean;

  @Column({ type: 'varchar', nullable: true })
  type: string;

  @Column({ type: 'varchar', nullable: true })
  title: string;

  @Column({ type: 'varchar', name: 'company_name', nullable: true })
  companyName: string;

  @Column({ type: 'varchar', nullable: true })
  location: string;

  @Column({ type: 'varchar', name: 'salary_range', nullable: true })
  salaryRange: string;

  @Column({ type: 'text', nullable: true })
  description: string;

  @Column({ type: 'text', name: 'how_to_apply', nullable: true })
  howToApply: string;

  @Column({ type: 'date', nullable: true })
  deadline: Date;

  @Column({ type: 'boolean', name: 'is_active', nullable: true })
  isActive: boolean;

  @CreateDateColumn({ type: 'timestamp', name: 'created_at' })
  createdAt: Date;
}

File: backend/src/main/java/com/service/backend/article/domain/entity/Job.java
package com.service.backend.article.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("jobs")
public class Job {

    @Id
    private Integer id;

    @Column("organization_id")
    private Integer organizationId;

    @Column("poster_member_id")
    private Integer posterMemberId;

    @Column("is_referral")
    @Builder.Default
    private Boolean isReferral = false;

    private String type;

    private String title;

    @Column("company_name")
    private String companyName;

    private String location;

    @Column("salary_range")
    private String salaryRange;

    private String description;

    @Column("how_to_apply")
    private String howToApply;

    private LocalDate deadline;

    @Column("is_active")
    @Builder.Default
    private Boolean isActive = true;

    @CreatedDate
    @Column("created_at")
    private LocalDateTime createdAt;
}
